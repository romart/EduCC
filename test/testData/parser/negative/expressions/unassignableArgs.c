int foo(int *pa, int *pb);

struct S1 {};
struct S2 {};

int fex(struct S1 s1, struct S2 s2);

int g = 10;

int bar(float f) {
  struct S1 s1;
  struct S2 s2;
  int arr[20];

  foo(g, g);
  foo(f, f);
  foo(&g, f);
  foo(&g, &g);
  foo(&f, &f);
  foo(s1, s2);
  foo(&g, s1);
  foo(&g, s2);
  foo(&s1, &s1);
  foo(&s2, &s2);
  foo(arr, arr);

  fex(f, f);
  fex(g, g);
  fex(&g, &g);
  fex(&f, &f);
  fex(s1, s1);
  fex(s2, s2);
  fex(s2, s1);
  fex(&s1, &s2);
  fex(&s1, &s1);
  fex(&s2, &s2);
  fex(&foo, &foo);
  fex(s1, s2);
  fex(arr, arr);
}
