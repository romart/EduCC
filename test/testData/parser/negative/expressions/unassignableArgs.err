test/testData/parser/negative/expressions/unassignableArgs.c:11:3: error: variable has incomplete type 'STRUCT S1'
test/testData/parser/negative/expressions/unassignableArgs.c:12:3: error: variable has incomplete type 'STRUCT S2'
test/testData/parser/negative/expressions/unassignableArgs.c:15:7: warning: incompatible integer to pointer conversion assigning to '*signed int' from 'signed int'
test/testData/parser/negative/expressions/unassignableArgs.c:15:10: warning: incompatible integer to pointer conversion assigning to '*signed int' from 'signed int'
test/testData/parser/negative/expressions/unassignableArgs.c:16:7: error: assigning to '*signed int' from incompatible type 'float'
test/testData/parser/negative/expressions/unassignableArgs.c:17:11: error: assigning to '*signed int' from incompatible type 'float'
test/testData/parser/negative/expressions/unassignableArgs.c:19:8: warning: incompatible pointer types assigning to '*signed int' from '*float'
test/testData/parser/negative/expressions/unassignableArgs.c:19:12: warning: incompatible pointer types assigning to '*signed int' from '*float'
test/testData/parser/negative/expressions/unassignableArgs.c:23:8: warning: incompatible pointer types assigning to '*signed int' from '*ERROR TYPE'
test/testData/parser/negative/expressions/unassignableArgs.c:23:13: warning: incompatible pointer types assigning to '*signed int' from '*ERROR TYPE'
test/testData/parser/negative/expressions/unassignableArgs.c:24:8: warning: incompatible pointer types assigning to '*signed int' from '*ERROR TYPE'
test/testData/parser/negative/expressions/unassignableArgs.c:24:13: warning: incompatible pointer types assigning to '*signed int' from '*ERROR TYPE'
test/testData/parser/negative/expressions/unassignableArgs.c:27:7: error: assigning to 'STRUCT S1' from incompatible type 'float'
test/testData/parser/negative/expressions/unassignableArgs.c:28:7: error: assigning to 'STRUCT S1' from incompatible type 'signed int'
test/testData/parser/negative/expressions/unassignableArgs.c:29:8: error: assigning to 'STRUCT S1' from incompatible type '*signed int'
test/testData/parser/negative/expressions/unassignableArgs.c:30:8: error: assigning to 'STRUCT S1' from incompatible type '*float'
test/testData/parser/negative/expressions/unassignableArgs.c:34:8: error: assigning to 'STRUCT S1' from incompatible type '*ERROR TYPE'
test/testData/parser/negative/expressions/unassignableArgs.c:35:8: error: assigning to 'STRUCT S1' from incompatible type '*ERROR TYPE'
test/testData/parser/negative/expressions/unassignableArgs.c:36:8: error: assigning to 'STRUCT S1' from incompatible type '*ERROR TYPE'
test/testData/parser/negative/expressions/unassignableArgs.c:37:8: error: assigning to 'STRUCT S1' from incompatible type 'C *{signed int (*signed int, *signed int)}'
test/testData/parser/negative/expressions/unassignableArgs.c:39:7: error: assigning to 'STRUCT S1' from incompatible type 'signed int[20]'
