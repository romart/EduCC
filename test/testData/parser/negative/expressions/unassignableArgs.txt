FILE test/testData/parser/negative/expressions/unassignableArgs.c
  STRUCT S1
----
  STRUCT S2
----
  FUN signed int foo 
    #0: *signed int pa
    #1: *signed int pb
----
  FUN signed int fex 
    #0: STRUCT S1 s1
    #1: STRUCT S2 s2
----
  signed int g = \
    10
----
  FUN signed int bar 
    #0: float f
  BEGIN
    ERROR TYPE s1
    ERROR TYPE s2
    signed int[20] arr
    foo((*signed int)(*g), (*signed int)(*g))
    foo(*f, *f)
    foo(g, *f)
    foo(g, g)
    foo((*signed int)f, (*signed int)f)
    foo((*signed int)(*s1), (*signed int)(*s2))
    foo(g, (*signed int)(*s1))
    foo(g, (*signed int)(*s2))
    foo((*signed int)s1, (*signed int)s1)
    foo((*signed int)s2, (*signed int)s2)
    foo((*signed int)arr, (*signed int)arr)
    fex(*f, *f)
    fex(*g, *g)
    fex(g, g)
    fex(f, f)
    fex((STRUCT S1)(*s1), (STRUCT S2)(*s1))
    fex((STRUCT S1)(*s2), (STRUCT S2)(*s2))
    fex((STRUCT S1)(*s2), (STRUCT S2)(*s1))
    fex(s1, s2)
    fex(s1, s1)
    fex(s2, s2)
    fex(foo, foo)
    fex((STRUCT S1)(*s1), (STRUCT S2)(*s2))
    fex(arr, arr)    
  END