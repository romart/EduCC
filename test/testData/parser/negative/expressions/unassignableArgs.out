test/testData/parser/negative/expressions/unassignableArgs.c:7:15: warning: incompatible integer to pointer conversion assigning to '*signed int' from 'signed int'
test/testData/parser/negative/expressions/unassignableArgs.c:10:15: warning: incompatible integer to pointer conversion assigning to '*signed int' from 'signed int'
test/testData/parser/negative/expressions/unassignableArgs.c:7:19: warning: incompatible pointer types assigning to '*signed int' from '*float'
test/testData/parser/negative/expressions/unassignableArgs.c:11:19: warning: incompatible pointer types assigning to '*signed int' from '*float'
test/testData/parser/negative/expressions/unassignableArgs.c:7:23: warning: incompatible pointer types assigning to '*signed int' from '*STRUCT S1'
test/testData/parser/negative/expressions/unassignableArgs.c:12:23: warning: incompatible pointer types assigning to '*signed int' from '*STRUCT S1'
test/testData/parser/negative/expressions/unassignableArgs.c:7:24: warning: incompatible pointer types assigning to '*signed int' from '*STRUCT S2'
test/testData/parser/negative/expressions/unassignableArgs.c:12:24: warning: incompatible pointer types assigning to '*signed int' from '*STRUCT S2'
